src/org/labkey/test/tests/AdminConsoleNavigationTest.java

```diff
@@ -71,7 +108,134 @@ public void testAdminNavTrails()
         }
 
         pagesMissingNavTrail.sort(Comparator.naturalOrder());
-        Assert.assertTrue("The following " + pagesMissingNavTrail.size() + " pages are missing Admin Console navtrails:\n" + String.join("\n", pagesMissingNavTrail), pagesMissingNavTrail.isEmpty());
+        assertTrue("The following " + pagesMissingNavTrail.size() + " pages are missing Admin Console navtrails:\n" + String.join("\n",
+                pagesMissingNavTrail), pagesMissingNavTrail.isEmpty());
+    }
+
+    @Test
+    public void testTroubleshooterLinkAccess()
+    {
+        Set<String> ignoredLinks = Collections.newSetFromMap(new CaseInsensitiveHashMap<>());
+        ignoredLinks.addAll(List.of(
+                "SignUp",                   //link shows up to the troubleshooter but throws 403 while accessing it.
+                "Notebook Settings",        //link shows up to the troubleshooter but throws 403 while accessing it.
+                "Puppeteer Service",        //link shows up to the troubleshooter but throws 403 while accessing it.
+                "Dump Heap",                // Undesired consequences
+                "Reset Site Errors",        // Undesired consequences
+                "Running Threads",          // Undesired consequences
+                "Profiler"                  //Profiler can be edited by the troubleshooter
+        ));
+        ShowAdminPage adminConsole = goToAdminConsole();
+        impersonate(TROUBLESHOOTER);
+        Map<String, String> linkHrefs = new LinkedHashMap<>();
+        List<WebElement> troubleshooterLinks = adminConsole.getAllAdminConsoleLinks();
+        for (WebElement link : troubleshooterLinks)
+            linkHrefs.put(link.getText(), link.getAttribute("href"));
+
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinks.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
```

Comment:
Nit:
You shouldn't need explicit reference to `TestLogger`

Reply:
Done